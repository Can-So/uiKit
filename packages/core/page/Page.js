import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
var Wrapper = styled.div.withConfig({
  displayName: "Page__Wrapper",
  componentId: "oj2x06-0"
})(["\n  display: flex;\n  flex-direction: column;\n  min-height: 100%;\n  width: 100%;\n"]);
var NavigationAndContent = styled.div.withConfig({
  displayName: "Page__NavigationAndContent",
  componentId: "oj2x06-1"
})(["\n  display: flex;\n  flex: 1 1 auto;\n"]);
var BannerContainer = styled.div.withConfig({
  displayName: "Page__BannerContainer",
  componentId: "oj2x06-2"
})(["\n  flex: 1 0 auto;\n  transition: height 0.25s ease-in-out;\n  height: ", "px;\n  position: relative;\n  width: 100%;\n  z-index: 3;\n"], function (props) {
  return props.isBannerOpen ? props.bannerHeight : 0;
});
var Banner = styled.div.withConfig({
  displayName: "Page__Banner",
  componentId: "oj2x06-3"
})(["\n  position: fixed;\n  width: 100%;\n"]);
var Navigation = styled.div.withConfig({
  displayName: "Page__Navigation",
  componentId: "oj2x06-4"
})(["\n  position: relative;\n  z-index: 2;\n"]);
var PageContent = styled.div.withConfig({
  displayName: "Page__PageContent",
  componentId: "oj2x06-5"
})(["\n  flex: 1 1 auto;\n  position: relative;\n  z-index: 1;\n  min-width: 0;\n"]);
var emptyTheme = {};

var Page =
/*#__PURE__*/
function (_Component) {
  _inherits(Page, _Component);

  function Page() {
    _classCallCheck(this, Page);

    return _possibleConstructorReturn(this, _getPrototypeOf(Page).apply(this, arguments));
  }

  _createClass(Page, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          isBannerOpen = _this$props.isBannerOpen,
          banner = _this$props.banner,
          navigation = _this$props.navigation,
          children = _this$props.children,
          bannerHeight = _this$props.bannerHeight;
      return React.createElement(ThemeProvider, {
        theme: emptyTheme
      }, React.createElement(Wrapper, null, this.props.banner ? React.createElement(BannerContainer, {
        "aria-hidden": isBannerOpen,
        isBannerOpen: isBannerOpen,
        bannerHeight: bannerHeight
      }, React.createElement(Banner, null, banner)) : null, React.createElement(NavigationAndContent, null, React.createElement(Navigation, null, navigation), React.createElement(PageContent, null, children))));
    }
  }]);

  return Page;
}(Component);

_defineProperty(Page, "displayName", 'AkPage');

_defineProperty(Page, "defaultProps", {
  isBannerOpen: false,
  bannerHeight: 52
});

export { Page as default };