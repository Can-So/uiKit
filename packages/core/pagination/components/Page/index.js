import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import React, { Component } from 'react';
import Button from '@findable/button';

var Page =
/*#__PURE__*/
function (_Component) {
  _inherits(Page, _Component);

  function Page() {
    _classCallCheck(this, Page);

    return _possibleConstructorReturn(this, _getPrototypeOf(Page).apply(this, arguments));
  }

  _createClass(Page, [{
    key: "render",
    value: function render() {
      return React.createElement(Button, _extends({}, this.props, {
        appearance: "subtle"
      }));
    }
  }]);

  return Page;
}(Component);

export { Page as default };