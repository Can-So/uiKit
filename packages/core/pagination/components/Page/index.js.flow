//@flow
import React, { Component } from 'react';
import Button, {
  type ButtonProps,
  type ButtonAppearances,
} from '@atlaskit/button';

type PagePropsType = $Diff<
  ButtonProps,
  {
    appearance?: ButtonAppearances,
    autoFocus: boolean,
    isDisabled: boolean,
    isLoading: boolean,
    spacing: 'compact' | 'default' | 'none',
    shouldFitContainer: boolean,
    type: 'button' | 'submit',
  },
>;

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
