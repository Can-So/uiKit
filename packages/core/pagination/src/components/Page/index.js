//@flow
import React, { Component } from 'react';
import Button from '@atlaskit/button';
import type { PagePropsType } from '../../types';

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
