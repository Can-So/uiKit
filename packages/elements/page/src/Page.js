// @flow
import React, { Component, type Node } from 'react';
import styled, { ThemeProvider } from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100%;
  width: 100%;
`;

const NavigationAndContent = styled.div`
  display: flex;
  flex: 1 1 auto;
`;

const BannerContainer = styled.div`
  flex: 1 0 auto;
  transition: max-height 0.25s ease-in-out;
  max-height: ${props =>
    props.isBannerOpen ? 52 : 0}px; /* 52 is line height (20) + 4*grid */
  position: relative;
  width: 100%;
  z-index: 3;
`;

const Banner = styled.div`
  position: fixed;
  width: 100%;
`;

const Navigation = styled.div`
  position: relative;
  z-index: 2;
`;

const PageContent = styled.div`
  flex: 1 1 auto;
  position: relative;
  z-index: 1;
`;

const emptyTheme = {};

type Props = {
  banner?: Node,
  children?: Node,
  isBannerOpen?: boolean,
  navigation?: Node,
};

export default class Page extends Component<Props, void> {
  static displayName = 'AkPage';

  static defaultProps = {
    isBannerOpen: false,
  };

  render() {
    return (
      <ThemeProvider theme={emptyTheme}>
        <Wrapper>
          {this.props.banner ? (
            <BannerContainer
              aria-hidden={this.props.isBannerOpen}
              isBannerOpen={this.props.isBannerOpen}
            >
              <Banner>{this.props.banner}</Banner>
            </BannerContainer>
          ) : null}
          <NavigationAndContent>
            <Navigation>{this.props.navigation}</Navigation>
            <PageContent>{this.props.children}</PageContent>
          </NavigationAndContent>
        </Wrapper>
      </ThemeProvider>
    );
  }
}
