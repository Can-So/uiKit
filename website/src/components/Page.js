// @flow
import styled from 'styled-components';
import { gridSize, math } from '@atlaskit/theme';

const containerWidth = {
  small: '480px',
  medium: '640px',
  large: '980px',
};

const PageContainer = styled.main`
  max-width: ${p =>
    containerWidth[p.width] ? containerWidth[p.width] : containerWidth.medium};
  margin: 2rem auto;
  padding: 0 2rem;
`;

export default PageContainer;

export const Title = styled.h1`
  margin-bottom: 1em;
`;

export const Section = styled.section`
  margin-top: 3em;

  p {
    line-height: 1.4em;
  }
`;

export const Intro = styled.p`
  font-size: ${math.multiply(gridSize, 2)}px;
  font-weight: 300;
  line-height: 1.4em;
`;
