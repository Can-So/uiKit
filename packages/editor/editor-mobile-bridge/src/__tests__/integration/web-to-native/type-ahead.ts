import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  editor,
  editable,
  getBridgeOutput,
  navigateOrClear,
  skipBrowsers as skip,
} from '../_utils';

BrowserTestCase(
  `type-ahead.ts: Sends correct typing events`,
  { skip: skip.concat('safari') },
  async (client: any) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'Some beautifully written words ');
    await browser.type(editable, '@idontexist');

    const typeAheadPayloads = await getBridgeOutput(
      browser,
      'typeAheadBridge',
      'typeAheadQuery',
    );

    expect(typeAheadPayloads).toMatchSnapshot();
  },
);

BrowserTestCase(
  `type-ahead.ts: Space sends dismiss type-ahead event.`,
  { skip },
  async (client: any) => {
    const browser = new Page(client);
    await navigateOrClear(browser, editor.path);
    await browser.waitForSelector(editable);

    // Space with no results dismisses typeahead
    await browser.type(editable, '@idontexist ');

    const typeAheadPayloads = await getBridgeOutput(
      browser,
      'typeAheadBridge',
      'dismissTypeAhead',
    );

    expect(typeAheadPayloads.length).toEqual(1);
  },
);

BrowserTestCase(
  `type-ahead.ts: Navigating away sends dismiss type-ahead event.`,
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any) => {
    const browser = new Page(client);
    await navigateOrClear(browser, editor.path);
    await browser.waitForSelector(editable);

    // Navigating cursor away from query mark should dismiss typeahead
    await browser.type(editable, ['Enter']);
    await browser.type(editable, 'word @sg');
    await browser.type(editable, [
      'ArrowLeft',
      'ArrowLeft',
      'ArrowLeft',
      'ArrowLeft',
    ]);

    const typeAheadPayloads = await getBridgeOutput(
      browser,
      'typeAheadBridge',
      'dismissTypeAhead',
    );

    expect(typeAheadPayloads.length).toEqual(1);
  },
);
