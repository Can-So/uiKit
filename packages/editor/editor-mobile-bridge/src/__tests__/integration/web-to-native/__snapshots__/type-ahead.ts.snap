// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type-ahead.ts type-ahead.ts: Sends correct typing events 1`] = `
Array [
  Object {
    "query": "",
    "trigger": "@",
  },
  Object {
    "query": "",
    "trigger": "@",
  },
  Object {
    "query": "i",
    "trigger": "@",
  },
  Object {
    "query": "i",
    "trigger": "@",
  },
  Object {
    "query": "id",
    "trigger": "@",
  },
  Object {
    "query": "id",
    "trigger": "@",
  },
  Object {
    "query": "ido",
    "trigger": "@",
  },
  Object {
    "query": "ido",
    "trigger": "@",
  },
  Object {
    "query": "idon",
    "trigger": "@",
  },
  Object {
    "query": "idon",
    "trigger": "@",
  },
  Object {
    "query": "idont",
    "trigger": "@",
  },
  Object {
    "query": "idont",
    "trigger": "@",
  },
  Object {
    "query": "idonte",
    "trigger": "@",
  },
  Object {
    "query": "idonte",
    "trigger": "@",
  },
  Object {
    "query": "idontex",
    "trigger": "@",
  },
  Object {
    "query": "idontex",
    "trigger": "@",
  },
  Object {
    "query": "idontexi",
    "trigger": "@",
  },
  Object {
    "query": "idontexi",
    "trigger": "@",
  },
  Object {
    "query": "idontexis",
    "trigger": "@",
  },
  Object {
    "query": "idontexis",
    "trigger": "@",
  },
  Object {
    "query": "idontexist",
    "trigger": "@",
  },
  Object {
    "query": "idontexist",
    "trigger": "@",
  },
]
`;
